components/style/properties/properties.mako.rs

```diff
@@ -1438,34 +1441,126 @@ impl PropertyId {
     }
 }
 
+<%
+    variants = []
+    for property in data.longhands:
+        if property.predefined_type and not property.is_vector and not property.is_size_type:
+            ty = "::values::specified::{}".format(property.predefined_type)
+        else:
+            ty = "longhands::{}::SpecifiedValue".format(property.ident)
+        if property.boxed:
+            ty = "Box<{}>".format(ty)
+        variants.append({
+            "name": property.camel_case,
+            "type": ty,
+            "doc": "`" + property.name + "`",
+        })
+    variants += [
+        {
+            "name": "CSSWideKeyword",
+            "type": "WideKeywordDeclaration",
+            "doc": "A CSS-wide keyword.",
+        },
+        {
+            "name": "WithVariables",
+            "type": "VariableDeclaration",
+            "doc": "An unparsed declaration.",
+        },
+        {
+            "name": "Custom",
+            "type": "CustomDeclaration",
+            "doc": "A custom property declaration.",
+        },
+    ]
+%>
+
 /// Servo's representation for a property declaration.
 #[cfg_attr(feature = "gecko", derive(MallocSizeOf))]
-#[derive(Clone, PartialEq)]
+#[derive(PartialEq)]
+#[repr(u8)]
```

Comment:
Doesn’t this have more than 256 variants?

Reply:
Yes. :D Just realised with the geckotry failing to build.